pennylane/transforms/combine_global_phases.py

```diff
@@ -0,0 +1,73 @@
+"""
+Provides a transform to combine all ``qml.GlobalPhase`` gates in a circuit into a single one applied at the end.
+"""
+
+import pennylane as qml
+from pennylane.tape import QuantumScript, QuantumScriptBatch
+from pennylane.transforms import transform
+from pennylane.typing import PostprocessingFn
+
+
+@transform
+def combine_global_phases(tape: QuantumScript) -> tuple[QuantumScriptBatch, PostprocessingFn]:
+    """Combine all ``qml.GlobalPhase`` gates into a single ``qml.GlobalPhase`` operation.
+
+    Circuits passed through this transform will be updated to include the same operations and
+    measurements of the original input other than ``qml.GlobalPhase``.
+    A single ``qml.GlobalPhase`` operation will be added at the end of the new output circuit.
+    The total global phase is computed as the algebraic sum of all the phases in the original circuit.
+
+    Args:
+        tape (QNode or QuantumTape or Callable): the input circuit to be transformed.
```

Comment:
We're moving away from the use of `QuantumTape`
```suggestion
        tape (QNode or QuantumScript or Callable): the input circuit to be transformed.
```